atmo/jobs/models.py

```diff
@@ -226,3 +213,53 @@ def delete(self, *args, **kwargs):
 
     def get_results(self):
         return self.provisioner.results(self.identifier, self.is_public)
+
+
+@python_2_unicode_compatible
+class SparkJobRun(EditedAtModel):
+
+    spark_job = models.ForeignKey(
+        SparkJob,
+        on_delete=models.CASCADE,
```

Comment:
Yeah, basically we wouldn't want to lose the data, even while we may want to allow users to delete the job. We currently don't delete job notebook files from S3 either, so there is already a cleanup required to be done eventually. In terms of history tracking, I'd rather have a bit too much info (especially since it doesn't contain any private info) that too little. Then once we built a graph in Redash against the table we can decide how far back we want to keep the data for sure, then then write the cleanup script if needed.